envparse.py

```diff
@@ -114,6 +114,22 @@ def __call__(self, var, default=NOTSET, cast=None, subcast=None,
         return value
 
     @classmethod
+    def track_path(cls, path="", tracking_table={}):
```

Comment:
The use of a mutable type as default value is a little dangerous. (see http://docs.python-guide.org/en/latest/writing/gotchas/). Recommend using `None` as default, and setting to `{}` inside function.